Add Cell Ranger mkfastq

lebovic opened the pull request:
Adds:
- Cell Ranger mkfastq

lebovic commented:
@bcai2 requesting your preliminary review; this is still untested

bcai2 reviewed (approved):
The naming of `_validate_and_prepare()` is confusing, esp. next to `_validate_and_prepare_inputs()`. Otherwise, LGTM

lebovic commented:
> The naming of `_validate_and_prepare()` is confusing, esp. next to `_validate_and_prepare_inputs()`. Otherwise, LGTM

Fixed by renaming `_validate_and_prepare()` back to `_validate_args()`, renamed `_validate_and_prepare_inputs()` to `_prepare_inputs()`, and moved the call of `_prepare_inputs()` out of `_validate_args()` to `run()`.

lebovic commented on toolchest_client/files/general.py:
```diff
@@ -69,6 +74,26 @@ def files_in_path(files):
+def compress_files_in_path(file_path):
+    """Returns a tarred and compressed file containing the contents of a directory.
+
+    WARNING: this is NOT thread-safe, as shutil.make_archive is not thread safe.
```
This means we can't run many compressions in parallel (for now)

bcai2 reviewed (approved):
LGTM, just bump the package version # and you should be good to merge.
